scripts/install_xcpng.py

```diff
@@ -62,24 +62,58 @@ def generate_answerfile(directory, installer, hostname_or_ip, type, hdd):
         elif type == 'upgrade':
             answerfile.write(f"""<?xml version="1.0"?>
 <installation mode="upgrade">
-    <existing-installation>sda</existing-installation>
+    <existing-installation>{hdd}</existing-installation>
```

Comment:
This change is not documented in the commit message and may even deserve a separate bugfix commit.